cpp/src/centrality/betweenness_centrality_impl.cuh

```diff
@@ -647,6 +647,40 @@ edge_betweenness_centrality(
                             do_expensive_check);
   }
 
+  std::optional<weight_t> scale_factor{std::nullopt};
+
+  if (normalized) {
+    weight_t n   = static_cast<weight_t>(graph_view.number_of_vertices());
+    scale_factor = n * (n - 1);
+  } else if (graph_view.is_symmetric())
+    scale_factor = weight_t{2};
+
+  if (scale_factor) {
+    if (graph_view.number_of_edges() > 2) {
```

Comment:
Alright. I was under the impression that the scale factor for edge betweenness centrality would be proportional to the number of edges as you can see from the comments I made. I will update that accordingly

Reply:
> Also, I think both this and the vertex betweenness check should be > 1

After having a close look at the C++ code, it should indeed be `graph_view.number_of_vertices() > 2` for `betweenness centrality` because in the worst case, we have a `scale_factor` where the denominator is (n-1)(n-2) which occurs when there is no endpoint. I believe you didn't see this [line](https://github.com/rapidsai/cugraph/blob/branch-23.08/cpp/src/centrality/betweenness_centrality_impl.cuh#L529)

Regarding edge betweenness centrality, it should be `graph_view.number_of_vertices() > 1` like you suggested.